src/stc/PlatWX.cpp

```diff
@@ -1983,433 +1997,743 @@ PRectangle Window::GetMonitorRect(Point pt) {
-        if (x != wxDefaultCoord) {
-            GetParent()->ClientToScreen(&x, NULL);
-        }
-        if (y != wxDefaultCoord) {
-            GetParent()->ClientToScreen(NULL, &y);
+        wxSTCPopupBase::~wxSTCPopupBase()
+        {
+            wxRect rect = GetRect();
+            GetParent()->ScreenToClient(&(rect.x), &(rect.y));
+            GetParent()->Refresh(false, &rect);
         }
-        wxPopupWindow::DoSetSize(x, y, width, height, sizeFlags);
-    }
 
-    // return position as if it were in client coords
-    virtual void DoGetPosition( int *x, int *y ) const wxOVERRIDE {
-        int sx, sy;
-        wxPopupWindow::DoGetPosition(&sx, &sy);
-        GetParent()->ScreenToClient(&sx, &sy);
-        if (x) *x = sx;
-        if (y) *y = sy;
-    }
+    #elif defined(__WXMSW__)
 
+        // Do not activate the window when it is shown.
+        bool wxSTCPopupBase::Show(bool show)
+        {
+            if ( !wxWindowBase::Show(show) )
+                return false;
 
-    bool Destroy() wxOVERRIDE {
-        if ( !wxPendingDelete.Member(this) )
-            wxPendingDelete.Append(this);
-        return true;
-    }
+            if ( show )
+            {
+                HWND hWnd = reinterpret_cast<HWND>(GetHandle());
+                if ( GetName() == "wxSTCCallTip" )
+                    ::AnimateWindow(hWnd, 25, AW_BLEND);
+                else
+                    ::ShowWindow(hWnd, SW_SHOWNA );
 
+                ::SetWindowPos(hWnd, HWND_NOTOPMOST, 0, 0, 0, 0,
+                               SWP_NOMOVE | SWP_NOSIZE | SWP_NOACTIVATE);
+            }
+            else
+                wxPopupWindow::Show(false);
 
-    int IconWidth() {
-        wxImageList* il = lv->GetImageList(wxIMAGE_LIST_SMALL);
-        if (il != NULL) {
-            int w, h;
-            il->GetSize(0, w, h);
-            return w;
+            return true;
         }
-        return 0;
-    }
 
+        // Do not activate in response to mouse clicks on this window.
+        bool wxSTCPopupBase::MSWHandleMessage(WXLRESULT *res, WXUINT msg,
+                                              WXWPARAM wParam, WXLPARAM lParam)
+        {
+            if ( msg == WM_MOUSEACTIVATE )
+            {
+                *res = MA_NOACTIVATE;
+                return true;
+            }
+            else
+                return wxPopupWindow::MSWHandleMessage(res, msg, wParam,lParam);
+        }
 
-    void SetDoubleClickAction(CallBackAction action, void *data) {
-        doubleClickAction = action;
-        doubleClickActionData = data;
-    }
+    #endif // __WXGTK__
 
+#else
 
-    void OnFocus(wxFocusEvent& event) {
-        GetParent()->SetFocus();
-        event.Skip();
-    }
+    wxSTCPopupBase::wxSTCPopupBase(wxWindow* parent):wxFrame()
+    {
+        // Make sure the frame is initially hidden. However, GTK+ will hide the
+        // frame initially and doesn't like trying to hide it before it's
```

Comment:
This is pretty bad actually, hiding any window before creating it is definitely supposed to work.

OTOH frames are also supposed to be created hidden under all platforms, so this call shouldn't be _necessary_ anywhere. It should't do any harm however neither...

Reply:
I just removed the call to Hide everywhere since I now know it's done by default everywhere.